### ISSUE TITLE
Crash on resize() with 1 pixel image and mode="reflect"

### ISSUE THREAD

**Issue by waleedka** [scikit-image/scikit-image#3001]
This code reproduces the issue:

```
image = np.zeros([1, 2])
image = skimage.transform.resize(image, (2, 4), order=1, mode="reflect")
```

This causes the Python interpreter to crash with a "floating point exception". I'm running this on Ubuntu 16.04, Python 3.5.2 and scikit-image 0.13.1.

Changing mode to `constant` or changing order to `0` avoids the problem.

**Comment by jni** [comment]
Well, this crash makes perfect sense but I don't know how to fix it. =P

@waleedka thanks for the report! Would you expect it to work or just to throw a meaningful error? I don't think reflect makes sense when the image is one-pixel-thick in one dimension.

Coincidentally, this is probably an argument in favour of making `mode='edge'` the default as proposed in #2990!

**Comment by waleedka** [comment]
> Would you expect it to work or just to throw a meaningful error? I don't think reflect makes sense when the image is one-pixel-thick in one dimension.

I'm no expert in the intricate details and edge cases of image resizing, but I would expect it to work rather than throw an error. And I don't see a problem with applying reflect on a 1-pixel image, it should simply repeat that pixel, similar to how Numpy handles it.
```
np.pad([[7]], 1, 'reflect')

# Out:
# array([[7, 7, 7],
#        [7, 7, 7],
#        [7, 7, 7]])
```

**Comment by jni** [comment]
Wow! That's very interesting, because that really doesn't jive with the definition of 'reflect'. From the `pad` docstring:

```
    >>> a = [1, 2, 3, 4, 5]
    >>> np.lib.pad(a, (2, 3), 'reflect')
    array([3, 2, 1, 2, 3, 4, 5, 4, 3, 2])
```

Note how the edge value (1) is not repeated: you are taking a reflection about that value to get the padding. In the length-1 case, you are trying to reflect from beyond the edge of the image to the other side, so you have no values to pull from...

**Comment by waleedka** [comment]
I see your point. You're right, it's an edge case. But I think repeating the 1 pixel is okay because the spirit of the `reflect` mode is to reduce discontinuities at the edges, and this achieves it.

**Comment by stefanv** [comment]
We should never segfault.

**Comment by stefanv** [comment]
```
0x00007fffe0418254 in __pyx_f_7skimage_7_shared_13interpolation_coord_map (__pyx_v_mode=82 'R', __pyx_v_coord=<optimized out>, __pyx_v_dim=1) at skimage/transform/_warps_cy.c:6998
6998	        __pyx_r = ((Py_ssize_t)(__pyx_v_cmax - (__pyx_v_coord % __pyx_v_cmax)));
```

**Comment by stefanv** [comment]
This is a valid bug; we should avoid segfaulting in our Cython code, and also catch this case and raise an error.

**Comment by ahojnnes** [comment]
See https://github.com/scikit-image/scikit-image/pull/3174

**Comment by JDWarner** [comment]
Closed by #3174
